src/Form/AppAnalyticsSettingsForm.php

```diff
@@ -81,7 +81,8 @@ public function getFormId() {
    * {@inheritdoc}
    */
   public function buildForm(array $form, FormStateInterface $form_state) {
-    $environments = $this->environmentController->getEntityIds();
+    // Remove portal environment from environment list.
+    $environments = array_diff($this->environmentController->getEntityIds(), ['portal']);
```

Comment:
Since  `'array_combine($environments, $environments),` is being called below (multiple times 🙄 ), it is cheaper if you do the `array_combine()` here and then call `unset($environments['portal']);`.

Also please leave a more detailed description about WHY the portal environment is removed (something like we can see in the original issue.)